### DISCUSSION TITLE
Use Pinia with stores from different locations

### DISCUSSION THREAD

**Discussion by evandor** [quasarframework/quasar#17209] [General - Components / Directives / etc] [answered]
The typical setup for pinia in quasar is to have the stores in the "src/stores" folder; creating a store with the cli will make sure I have the index.ts file (and maybe the store-flag.d.ts file too?). This setup works perfectly, but when I want to create an additional store at another location (say, "src/somefeature/stores"), although everything seems ok, the store will not work (I mean it is not reactive).

So, before spending too much time figuring out how this might work, I'd like to ask if this is supported at all? Is pinia in quasar meant to be used in such a way? Or should the stores all be placed in that fixed "src/stores" folder?

Thanks so much!

**Comment by yusufkandemir** [comment]
`src/stores/index.ts` and `src/stores/store-flag.d.ts` are mandatory, and it's recommended to define the stores there just for organization purposes. You can update `quasar.config file > sourceFiles > store` to use a path other than `src/stores` to store those files. Other than that, you can put the stores wherever you want. Neither Quasar nor Pinia relies on the file paths to define how stores work.

**Reply by evandor** [reply]
Thank you for your answer. If I understand correctly, I can redefine where to look for the stores in quasar.config.js. However, it seems that it's not possible to define an array of folders to place the stores in, is that right?

The reason I'm interested in this is that I want to organize my project structure by domain elements (following Domain-Driven Design principles), rather than by technical aspects. For example, instead of "src/stores/ordersStore.ts" and "src/stores/itemsStore.ts", I'd prefer a structure like "src/orders/store/ordersStore.ts" and "src/items/store/itemsStore.ts".

From the documentation and your response, it appears this approach isn't feasible.

**Reply by yusufkandemir** [reply, marked as answer]
I think you got me wrong. That path is only used for the `index.ts` and `store-flag.d.ts`. For defining the stores, you can use it, but not limited to this path. So, you should already be able to achieve what you said. If as you said, there is a reactivity problem, then you can share a reproduction so we can inspect it.

**Reply by evandor** [reply]
Ok, I tried to create a simple reproduction project and was not able to reproduce the issue. I'll investigate further to find out what is the root cause for the issue in my current project - but now I know that it _should_ work :) 

Thanks a lot for your support, all the best!
